### DISCUSSION TITLE
Best way to highlight json logs

### DISCUSSION THREAD

**Discussion by tony-sol** [sharkdp/bat#2732] [Q&A] [answered]
I use `bat --paging=never -l log` to highlight logs output from docker containers, and have log record as valid json struct.
Is there a best way to combine language arguments, which would say that all stdin stream should be processed like `-l log` and lines from stream like `-l json`?

**Comment by keith-hall** [comment, marked as answer]
This isn't really possible without editing the syntax definition, which is basically just a YAML file with regex patterns. https://github.com/sharkdp/bat/blob/master/assets/syntaxes/02_Extra/log.sublime-syntax.

Likely you'd want to add a new `- match` to the `main_without_log_level_line` context to look for `{` and `embed` the JSON syntax and pop at EOL.
You should be able to experiment by copying the `log.sublime-syntax`  file to your bat config folder and following the instructions https://github.com/sharkdp/bat/tree/master#adding-new-syntaxes--language-definitions

If you can share some example log lines, I might be able to spare some time to work on it at some point.

**Reply by tony-sol** [reply]
Well, it turned out that in my case the problem is different - vscode theme has same color for json string values and object keys.
Using other theme, like TwoDark, makes output colorized exactly as i need

<img width="1582" alt="json-formatted log" src="https://github.com/sharkdp/bat/assets/33671815/6bf607bd-208c-4326-9855-78168c1496f6">

**Reply by keith-hall** [reply]
Ah nice, thanks for sharing and glad you got it working the way you want 🙂 probably I should update the highlighted test file for the log syntax to include some JSON like that, then I might have spotted it myself 😉
